src/components/Nav/Nav.js

```diff
@@ -1,10 +1,113 @@
 import React from 'react';
+import { withRouter } from 'react-router-dom';
+import { Link } from 'react-router-dom';
+import Button from '../Button/Button';
 import './Nav.scss';
 
 class Nav extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      isOpen: false,
+      userLists: {},
+    };
+  }
+
+  hadleUserData() {
+    console.log('handleUserData실행');
+
+    fetch('/data/userData.json', {
+      method: 'GET', // GET method는 기본값이라서 생략이 가능합니다.
+    }) // 예시코드에서는 이해를 돕기 위해 명시적으로 기입해뒀습니다.
+      .then(res => res.json())
+      .then(data =>
+        this.setState({
+          userLists: data,
+        })
+      );
+  }
+
+  componentDidMount() {
+    this.hadleUserData();
+  }
+
+  openMenu = () => {
+    if (this.state.isOpen === false) {
+      this.setState({
+        isOpen: true,
+      });
+    } else {
+      this.setState({
+        isOpen: false,
+      });
+    }
+  };
```

Comment:
1. 메뉴를 열고 닫는 기능이 모두 들어있으므로 함수명을 openMenu에서 handleMenu와 같이 변경하면 더 좋겠네요!
2. false일 때는 true로 true일 때는 false로 값이 변경되야 하므로 결국 현재의 상태값이 반대로 바뀌면 되겠죠? 아래 코드 참고해주세요!

```suggestion
  handleMenu = () => {
    this.setState({
      isOpen : !this.state.isOpen
    });
  };
```

Reply:
헉..도현님 이렇게 깔끔 해 질수 있군요! 앞으로도 이런식으로 작성하도록 하겠습니다!!!